tools/lint-docstring

```diff
@@ -35,9 +37,64 @@ DESC_FOR_NO_FILE_FOLDERS = {
 FOLDERS_TO_EXCLUDE = ["__pycache__"]
 
 
-def main() -> None:
-    create_file_overview_doc()
-    print(f"Generated {DEVELOPER_DOC_NAME} successfully.")
+def main(fix_file: bool) -> None:
+    if fix_file:
+        create_file_overview_doc()
+        print(in_color("green", "Generated " + DEVELOPER_DOC_NAME + " successfully."))
+    else:
+        lint_file_overview()
+        print(in_color("green", DEVELOPER_DOC_NAME + " has been linted successfully."))
+
+
+def lint_file_overview() -> None:
+    """
+    Used for linting of the developer-file-overview.md document
+    """
+    folder_file_docstring = generate_folder_file_docstrings_dict()
+    computed_doc_dict = extract_docstrings_from_file_overview()
+    missing_entity = ""
+    try:
+        for folder in folder_file_docstring:
+            for file in folder_file_docstring[folder]:
+                missing_entity = file
+                assert (
+                    computed_doc_dict[folder][file]
+                    == folder_file_docstring[folder][file]
+                )
+        # zulipterminal/themes and zulipterminal/scripts are linted using DESC_FOR_NO_FILE_FOLDERS
+        for folder_name in DESC_FOR_NO_FILE_FOLDERS:
+            missing_entity = folder_name
+            assert (
+                computed_doc_dict[folder_name]["no_file_present"]
+                == DESC_FOR_NO_FILE_FOLDERS[folder_name]
+            )
```

Comment:
These assert statements do work, with the downside that they only show one error result, even if there are more than one.